core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/DatastoreServerBootstrap.java

```diff
@@ -147,9 +150,26 @@ public void run(SnowOwlConfiguration configuration, Environment env, IProgressMo
 			initializeJobSupport(env, configuration);
 			initializeRepositories(configuration, env);
 			initializeRequestSupport(env, configuration.getModuleConfig(RepositoryConfiguration.class).getNumberOfWorkers());
+			verifyRepositories(env);
 		}
 	}
 	
+	@SuppressWarnings("deprecation")
+	private void verifyRepositories(Environment env) {
+		final DefaultRepositoryManager repositories = (DefaultRepositoryManager) env.service(RepositoryManager.class);
+		Optional<InternalRepository> inConsistentRepository = repositories.repositories()
+			.stream()
+			.filter(repository -> repository instanceof InternalRepository).map(InternalRepository.class::cast)
+			.filter(repository -> !repository.isConsistent(BranchPathUtils.createMainPath())) 
+			/* XXX: or should we check all the branches, not just the main??*/
```

Comment:
Good question! 😄 The problem is that in a production environment, there are _lots_ of branches. There is no easy way of retrieving commit infos from CDO across branches either, and even if it were, it might take up a lot of memory so you'd have to do it one at a time, in which case it would be slow instead of memory-consuming. 

We can keep the check on `MAIN` only for now.